### PR TITLE
Read aggregate public key from genesis block header

### PR DESCRIPTION
**Pull request by Naviabheeman**
In this pull request 

- The aggregate public key is read from genesis block header.   
- Command line argument signblockpubkey is removed.
- Other blocks do not have the aggregate pubkey.
- get block and getblockheader RPCs can list aggregate public key if it was present in the block header

Unit tests and Functional tests are updated to so the same.

### CODE REVIEW

File: src/tapyrus-genesis.cpp [hunk 1]
```diff
@@ -60,6 +58,7 @@ static int AppInit(int argc, char* argv[])
     // Check for -testnet or -regtest parameter (Params() calls are only valid after this clause)
     try {
         SelectParams(gArgs.GetChainName());
+        const_cast<CChainParams&>(Params()).ReadAggregatePubkey(ParseHex(gArgs.GetArg("-signblockpubkey", "")));
```
[the remark this answers is not in the document]
Comment (Naviabheeman) [reply src/tapyrus-genesis.cpp]: I moved this code down.

File: src/rpc/mining.cpp [hunk 2]
```diff
@@ -117,13 +95,14 @@ static UniValue generatetoaddress(const JSONRPCRequest& request)
             "\nMine blocks immediately to a specified address (before the RPC call returns)\n"
             "\nArguments:\n"
             "1. nblocks      (numeric, required) How many blocks are generated immediately.\n"
-            "2. address      (string, required) The address to send the newly generated coin to.\n"
-            "3. [private keys] (hex string array, required) to sign the generated blocks.\n"
+            "2. address      (string, required) The address to send the newly generated bitcoin to.\n"
+            "3. private key  (hex string array, required) addregate private key to sign the generated blocks.\n"
```
[the remark this answers is not in the document]
Comment (Naviabheeman) [reply src/rpc/mining.cpp]: fixed both generatetoaddress and generate rpcs

File: src/rpc/mining.cpp [hunk 3]
```diff
@@ -117,13 +95,14 @@ static UniValue generatetoaddress(const JSONRPCRequest& request)
             "\nMine blocks immediately to a specified address (before the RPC call returns)\n"
             "\nArguments:\n"
             "1. nblocks      (numeric, required) How many blocks are generated immediately.\n"
-            "2. address      (string, required) The address to send the newly generated coin to.\n"
-            "3. [private keys] (hex string array, required) to sign the generated blocks.\n"
+            "2. address      (string, required) The address to send the newly generated bitcoin to.\n"
+            "3. private key  (hex string, required) aggregate private key in WIF to sign the generated blocks.\n"
```
Comment (rantan) [inline src/rpc/mining.cpp]: Above document is wrong. Private key should be HEX string  not WIF.

**Comment by rantan** [comment]
`doc/tapyrus/signedblocks` should be updated. Can you update after this PR is finished?

**Pull request merged**